x-pack/plugins/security/public/cloud_ui_tweaks/maybe_add_cloud_links.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { catchError, defer, filter, map, of } from 'rxjs';
+
+import type { CloudStart } from '@kbn/cloud-plugin/public';
+import type { ChromeStart } from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+
+import type { AuthenticationServiceStart, SecurityNavControlServiceStart } from '..';
+import { createUserMenuLinks } from './user_menu_links';
+
+export interface MaybeAddCloudLinksDeps {
+  authc: AuthenticationServiceStart;
+  chrome: ChromeStart;
+  cloud: CloudStart;
+  navControlService: SecurityNavControlServiceStart;
+}
+
+export function maybeAddCloudLinks({
+  authc,
+  chrome,
+  cloud,
+  navControlService,
+}: MaybeAddCloudLinksDeps): void {
+  if (cloud.isCloudEnabled) {
+    defer(() => authc.getCurrentUser())
+      .pipe(
```

Comment:
We'll see what the security team thinks about it, but I feel like this is in the wrong place. I would have put that in some kind of 'bridge' plugin in charge of registering all cloud-related stuff (e.g a `cloud-environment-setup` plugin). Following this logic, we could also have put the `home` plugin cloud environment setup (that this PR moves from `cloud` to `home`) in the same plugin.

Reply:
it makes a lot of sense. I can rework this.

My main driver to move it to the `home` and `security` plugins, respectively, was that other plugins have custom extra logic under `cloud.isCloudEnabled`, so extending the user menu (registered by the `security` plugin) with extra cloud links seemed to follow a similar approach.

Having said that, I'm ++ to keeping as much cloud-specific logic as possible in cloud-specific plugins.

Reply:
Moved!